### PR TITLE
Fixes NullPointerException during update. Fixes #1484

### LINKED ISSUE #1484: 0.13.6-M1 NullPointerException during update
While trying out the 0.13.6-M1 for #1477 I found a `NullPointerException`. After some debugging, I narrowed down the problem to the following two lines:

``` scala
conflictManager := ConflictManager.strict.copy(organization = "^(?!org\\.scala-lang).*$")

libraryDependencies += "org.specs2" %% "specs2" % "2.3.10-scalaz-7.1.0-M6" % "test"
```

I estimate that we might be the only company in the world with a configuration anything similar to this.

### PR DESCRIPTION
**Pull request by eed3si9n**

### CODE REVIEW

File: ivy/src/main/scala/sbt/IvyRetrieve.scala [hunk 1]
```diff
@@ -71,25 +78,29 @@ object IvyRetrieve {
...
-    val branch = Option(revId.getBranch)
+    val branch = nonEmptyString(revId.getBranch)
     val (status, publicationDate, resolver, artifactResolver) = dep.isLoaded match {
       case true =>
-        (Option(dep.getDescriptor.getStatus),
+        (nonEmptyString(dep.getDescriptor.getStatus),
           Some(new ju.Date(dep.getPublication)),
-          Option(dep.getModuleRevision.getResolver.getName),
-          Option(dep.getModuleRevision.getArtifactResolver.getName))
+          nonEmptyString(dep.getModuleRevision.getResolver.getName),
```
Comment (jsuereth) [inline ivy/src/main/scala/sbt/IvyRetrieve.scala:86]: dang, that's a lot of guarding.

File: main/src/main/scala/sbt/Defaults.scala [hunk 2]
```diff
@@ -1286,7 +1286,16 @@ object Classpaths {
             case (_, Some(out)) if uptodate(inChanged, out) => out
             case _ => work(in)
           }
-          outCache(in)
+          try {
+            outCache(in)
+          } catch {
+            case e: NullPointerException =>
```
Comment (jsuereth) [inline main/src/main/scala/sbt/Defaults.scala:1292]: You just want to catch null pointer exception?  Also, maybe you should log the exception.

**Comment by jsuereth** [comment]
Minor comment on exception handling, otherwise LGTM

File: main/src/main/scala/sbt/Defaults.scala (hunk 2 above)
Comment (eed3si9n) [inline main/src/main/scala/sbt/Defaults.scala:1292]: amended the commit and -f pushed with log.trace. See line 1299.

**Comment by jsuereth** [comment]
LGTM

**Pull request merged**